pennylane/transforms/optimization/pattern_matching.py

```diff
@@ -273,6 +275,56 @@ def pattern_matching(circuit_dag, pattern_dag):
 
     Returns:
         list(Match): the list of maximal matches.
+
```

Comment:
How about having a `seealso` link here to `pattern_matching_optimization`?

Reply:
@antalszava I will add that!